executor/builder.go

```diff
@@ -4070,3 +4083,82 @@ func (b *executorBuilder) buildTableSample(v *plannercore.PhysicalTableSample) *
 	}
 	return e
 }
+
+func (b *executorBuilder) buildCTE(v *plannercore.PhysicalCTE) Executor {
+	// 1. Build seedPlan.
+	seedExec := b.build(v.SeedPlan)
+	if b.err != nil {
+		return nil
+	}
+
+	// 2. Build iterIntTbl.
+	chkSize := b.ctx.GetSessionVars().MaxChunkSize
+	tps := seedExec.base().retFieldTypes
+	iterOutTbl := cteutil.NewStorageRC(tps, chkSize)
+	if err := iterOutTbl.OpenAndRef(); err != nil {
+		b.err = err
+		return nil
+	}
+
+	var resTbl cteutil.Storage = nil
+	var iterInTbl cteutil.Storage = nil
+	storageMap, ok := b.ctx.GetSessionVars().StmtCtx.CTEStorageMap.(map[int]*CTEStorages)
```

Comment:
I wonder why store `CTEStorageMap` in `StmtCtx` rather than `executorBuilder`

Reply:
Because **subquery** use different executorBuilder. The cteutil.Storage cannot be shared between subquery and main query if we put the map in executorBuilder.